### PR TITLE
Remove spoon implementation for backend csv

### LINKED ISSUE #2099: Change the implementation of Backend/Core/Engine/Csv away from spoon library
## Type

- Enhancement

## Problem description

We are using Spoon csv, since spoon is deprecated we should swap it out

## Expected behavior / Proposed solutions
Keep the same external api but let it work on a different library or implementation

### PR DESCRIPTION
**Pull request by Senjutsuu**
## Type
- Enhancement

## Resolves the following issues
fixes #2099 

## Pull request description
This pull request will no longer implement spoon library for CSV handling in Backend/Core/Engine/Csv.
Instead we will use the Symfony CSV-encoder and serializer.

### CODE REVIEW

File: src/Backend/Core/Engine/Csv.php [hunk 1]
```diff
@@ -54,6 +60,7 @@ public static function outputCSV(
         );
     }
 
+    // we are not using the line ending for now.
```
Comment (WouterSioen) [inline src/Backend/Core/Engine/Csv.php]: Not sure if it's safe to drop this feature? Will all software handling csv's on multiple OS'es still be able to handle this?

Comment (StijnVrolijk) [reply src/Backend/Core/Engine/Csv.php]: @WouterSioen Symfony's [CSV encoder](https://api.symfony.com/3.3/Symfony/Component/Serializer/Encoder/CsvEncoder.html) uses `fputcsv` to write CSV files so I assume it relies on PHP to properly detect the correct line endings. Not too sure if that's a good idea, though.

@Senjutsumo can you test on some different machines (OS X, Windows (with Excel), ...) to see if the CSV encoder always write readable files?

If it doesn't we could look at a CSV generating package on packagist.

**Comment by carakas** [comment]
I would use https://csv.thephpleague.com instead

**Comment by carakas** [comment]
replaced by #3160

**Pull request closed without merging**